internal/kubernetes/operator/project/project.go

```diff
@@ -339,87 +340,92 @@ func buildIntegrations(intProvider store.IntegrationLister, projectID, targetNam
 	var intSecrets []*corev1.Secret
 
 	for _, list := range integrations.Results {
-		secret := secrets.NewAtlasSecret(fmt.Sprintf("%s-integration-%s", projectID, strings.ToLower(list.Type)),
+		iType := getIntegrationType(reflect.ValueOf(list))
+		secret := secrets.NewAtlasSecret(fmt.Sprintf("%s-integration-%s", projectID, strings.ToLower(iType)),
 			targetNamespace, map[string][]byte{secrets.PasswordField: []byte("")}, dictionary)
 
 		integration := operatorProject.Integration{
-			Type: list.Type,
+			Type: iType,
 		}
 		secretRef := common.ResourceRefNamespaced{
 			Name:      resources.NormalizeAtlasName(secret.Name, dictionary),
 			Namespace: targetNamespace,
 		}
-		switch list.Type {
+		switch iType {
 		case "PAGER_DUTY":
 			integration.ServiceKeyRef = secretRef
 			if includeSecrets {
-				secret.Data[secrets.PasswordField] = []byte(list.ServiceKey)
+				secret.Data[secrets.PasswordField] = []byte(list.PagerDuty.ServiceKey)
 			}
 		case "SLACK":
-			integration.TeamName = list.TeamName
+			integration.TeamName = list.Slack.GetTeamName()
 			integration.APITokenRef = secretRef
 			if includeSecrets {
-				secret.Data[secrets.PasswordField] = []byte(list.APIToken)
+				secret.Data[secrets.PasswordField] = []byte(list.Slack.ApiToken)
 			}
-		case "DATADOG", "OPS_GENIE":
-			integration.Region = list.Region
+		case "DATADOG":
+			integration.Region = list.Datadog.GetRegion()
 			integration.APIKeyRef = secretRef
 			if includeSecrets {
-				secret.Data[secrets.PasswordField] = []byte(list.APIKey)
+				secret.Data[secrets.PasswordField] = []byte(list.Datadog.ApiKey)
 			}
-		case "FLOWDOCK":
-			integration.FlowName = list.FlowName
-			integration.OrgName = list.OrgName
-			integration.APITokenRef = secretRef
+		case "OPS_GENIE":
+			integration.Region = list.OpsGenie.GetRegion()
+			integration.APIKeyRef = secretRef
 			if includeSecrets {
-				secret.Data[secrets.PasswordField] = []byte(list.APIToken)
+				secret.Data[secrets.PasswordField] = []byte(list.OpsGenie.ApiKey)
 			}
+		//case "FLOWDOCK":
```

Comment:
IMHO those are outdated. It would be worth to create jira outside the project to review our integrations in the atlas cli.